### ISSUE TITLE
warn(dead_code) incorrectly on derive(Debug)

### ISSUE THREAD

**Issue by yonilerner** [rust-lang/rust#135422]
I tried this code:

```rust
#[derive(Debug)]
enum Foo {
    Bar(String)
}

fn main() {
    let foo = Foo::Bar("hi".to_string());
    println!("{:?}", foo);
}
```

I expected to see this happen: The code should output `Bar("hi")` with no compiler warnings.

Instead, this happened: The code does output `Bar("hi")`, but there is the following compiler warning:
```
warning: field `0` is never read
 --> src/main.rs:3:9
  |
3 |     Bar(String)
  |     --- ^^^^^^
  |     |
  |     field in this variant
  |
  = note: `Foo` has a derived impl for the trait `Debug`, but this is intentionally ignored during dead code analysis
  = note: `#[warn(dead_code)]` on by default
help: consider changing the field to be of unit type to suppress this warning while preserving the field numbering, or remove the field
  |
3 |     Bar(())
  |         ~~

warning: `playground` (bin "playground") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.83s
     Running `target/debug/playground`
```

### Meta
Heres a playground link: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=9b6d2f66e6c8037781a45025529fdb14

At the current moment of testing, playground is using:
1.86.0-nightly (2025-01-11 eb54a50837ad4bcc9842)

**Comment by theemathas** [comment]
Not a bug. The error message you posted contains this explanation:

> note: `Foo` has a derived impl for the trait `Debug`, but this is intentionally ignored during dead code analysis

**Comment by jieyouxu** [comment]
This is indeed intentional and a duplicate of #88900.
